method.rb

```diff
@@ -0,0 +1,122 @@
+# frozen_string_literal: true
+
+module Enumerable
+  def my_each
+    return true unless block_given?
+
+    (0...length).each do |i|
+      yield(self[i])
+    end
+  end
```

Comment:
1. my_each returns an Enumerator if no block is given

`[1,2,3,4].my_each` should return `#<Enumerator: [1, 2, 3, 4]:each>`

this can be with `return to_enum`